src/components/CourseInfo/CourseInfo.jsx

```diff
@@ -1,31 +1,23 @@
-import React, { useEffect, useState } from 'react';
+// src/components/CourseInfo/CourseInfo.js
+import React, { useEffect } from 'react';
 import { useParams, useNavigate } from 'react-router-dom';
 import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchCourseById } from '../../store/courses/actions';
 import './CourseInfo.css';
-import { ENV } from '../../constants';
 
-const CourseInfo = ({ authors }) => {
+const CourseInfo = () => {
   const { courseId } = useParams();
   const navigate = useNavigate();
-  const [course, setCourse] = useState(null);
+  const dispatch = useDispatch();
+  const course = useSelector((state) => state.courses.find((course) => course.id === courseId));
```

Comment:
Ideally these selectors should be placed under `selectors.js` file for a specific store slice as per [recommendations](https://d17btkcdsmqrmh.cloudfront.net/react-fundamentals/docs/module-3/home-task/good_practices)